src/pubmetric/api_controller.py

```diff
@@ -0,0 +1,77 @@
+from fastapi import FastAPI, UploadFile, File
+from pydantic import BaseModel
+from typing import Dict
+import tempfile
+import os 
+import json
+from datetime import datetime
+
+from pubmetric.metrics import *
+from pubmetric.workflow import parse_cwl_workflows
+from pubmetric.network import create_citation_network
+
+app = FastAPI()
+
+class ScoreResponse(BaseModel):
+    workflow_scores: Dict
+
+class GraphRequest(BaseModel):
+    topic_id: str
+    test_size: str
+
+@app.post("/score_workflows/", response_model=ScoreResponse)
+async def score_workflows(file: UploadFile = File(None)):
+    path_to_data = "out_202407041439" # where does one store things later?  
+    metadata_filename = "../tool_metadata_topic_0121_20240703.json"
+
+    graph = await create_citation_network(inpath=path_to_data) # should I maybe change the name so it does not contain the date, so you have to look in the file instead, or perhaps it is regenerated outside of the package entrirely
```

Comment:
why is the file stored locally? It could be just read without storing it right? I assume that you will receive 1 workflow in CWL and you can verify and parse it without storing it locally, something similar to:

```
async def score_workflows(file: UploadFile = File(...)):
    # Read the contents of the uploaded file
    contents = await file.read()
    
    # Parse the YAML content as a CWL workflow (this tests whether the file is a YAML file)
    try:
        cwl_content = yaml.safe_load(contents)
    except yaml.YAMLError as e:
        return {"error": "Invalid YAML file", "details": str(e)}
    
    # Extract information from the CWL workflow
    extracted_info = extract_info_from_cwl(cwl_content)